streams/src/test/java/org/apache/kafka/streams/KafkaStreamsTest.java

```diff
@@ -616,7 +616,7 @@ public void shouldThrowExceptionSettingUncaughtExceptionHandlerNotInCreateState(
         final KafkaStreams streams = new KafkaStreams(getBuilderWithSource().build(), props, supplier, time);
         streams.start();
         try {
-            streams.setUncaughtExceptionHandler(null);
+            streams.setUncaughtExceptionHandler((Thread.UncaughtExceptionHandler) null);
```

Comment:
Should we also test the behavior of setting the new handler to `null`? And actually, why is `null` allowed now, but it wasn't before?

Reply:
The test isn't really related to null its check to see if the client is in the correct state to set the handler. 

The reason I added the cast is because the complier complains otherwise. I added the same test for the new handler below.

Reply:
Ah, gotcha. Thanks.